src/web/app/pages-session/session-result-page/session-result-page.component.ts

```diff
@@ -43,41 +48,113 @@ export class SessionResultPageComponent implements OnInit {
   questions: QuestionOutput[] = [];
   formattedSessionOpeningTime: string = '';
   formattedSessionClosingTime: string = '';
+  personName: string = '';
+  courseId: string = '';
+  feedbackSessionName: string = '';
+  regKey: string = '';
 
-  constructor(private feedbackSessionsService: FeedbackSessionsService, private route: ActivatedRoute,
-      private timezoneService: TimezoneService, private statusMessageService: StatusMessageService) {
+  private backendUrl: string = environment.backendUrl;
+
+  constructor(private feedbackSessionsService: FeedbackSessionsService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private timezoneService: TimezoneService,
+              private navigationService: NavigationService,
+              private authService: AuthService,
+              private studentService: StudentService,
+              private statusMessageService: StatusMessageService) {
     this.timezoneService.getTzVersion(); // import timezone service to load timezone data
   }
 
   ngOnInit(): void {
     this.route.queryParams.subscribe((queryParams: any) => {
-      const { courseid: courseId, fsname: feedbackSessionName }: Record<string, string> = queryParams;
-      this.feedbackSessionsService.getFeedbackSession({
-        courseId,
-        feedbackSessionName,
+      this.courseId = queryParams.courseid;
+      this.feedbackSessionName = queryParams.fsname;
+      this.regKey = queryParams.key || '';
+
+      const nextUrl: string = `${window.location.pathname}${window.location.search}`;
+      this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
+        if (this.regKey) {
+          const intent: Intent = Intent.STUDENT_RESULT;
+          this.authService.getAuthRegkeyValidity(this.regKey, intent).subscribe((resp: RegkeyValidity) => {
+            if (resp.isValid) {
+              if (auth.user) {
+                // The logged in user matches the registration key; redirect to the logged in URL
+
+                const loggedInUrl: string = window.location.href.replace('/web/sessions', '/web/student/sessions')
```

Comment:
Actually just curious what's stopping them from manually navigating to `/web/student/sessions`? Is there a separate auth check on that page? I tried to explore the code base but couldn't figure out how page navigation and authentication works in Angular :")

Reply:
Front-end: Student pages, instructor pages, and admin pages are all guarded and require login.
Back-end: Only if the intent is right and the authentication is right (based on registration key or Google ID information stored in cookie) then data will be provided.

So it is secure in both aspects.